### ISSUE TITLE
Missing new version link on dataset landing pages pointing to a resource map PID

### ISSUE THREAD

**Issue by drkrynstrng** [NCEAS/metacatui#725]
After updating a data package with provenance chains using `dataone::uploadDataPackage`, the new version link does not show up at the top of the page when trying to view the old resource map and it automatically directs to the new version. The function appears to be working correctly and the system metadata is also correct with respect to version chains.

A test example:

Original resource map: https://test.arcticdata.io/#view/resource_map_urn:uuid:c62c0697-5165-40be-a905-7d69bca75ad0

Updated resource map: https://test.arcticdata.io/#view/resource_map_urn:uuid:bcbfff40-5d6e-4f50-9e3a-3177aca3d3fe

Also note that the metadata PID in the General > Identifier field is incorrect.

Workflow:
[gh_issue_725.txt](https://github.com/NCEAS/metacatui/files/2246884/gh_issue_725.txt)

**Comment by jeanetteclark** [comment]
I am seeing this on production as well: 
https://knb.ecoinformatics.org/#view/urn:uuid:186d41fc-1e38-4e7d-96da-3fcf558b2aba
https://knb.ecoinformatics.org/#view/doi:10.5063/F1TD9VHH

version chain is intact in system metadata, but the new version link doesn't show up.

The key thing here seems to be the presence of prov in the dataset. @drkrynstrng is going to post his workflow momentarily

**Comment by drkrynstrng** [comment]
See the attached script for the workflow: [gh_issue_725.txt](https://github.com/NCEAS/metacatui/files/2246787/gh_issue_725.txt)

**Comment by jeanetteclark** [comment]
also - let me know if this is not the right repo. I _think_ this is a metacatUI issue but could be wrong

**Comment by laurenwalker** [comment]
Yep, this is a MetacatUI issue. The data package landing page should still render the obsoleted resource map if you navigate to the landing page via the obsoleted resource map id.

Thanks for the bug report Derek and Jeanette!

**Comment by robyngit** [comment]
The #2567 refactor loads file table that matches the given resource map when navigating to the view page using the resMap ID. So opening the viewer with a specific map no longer immediately replaces its contents with the newer package.

However, the newer version check looks at the version chain of the metadata doc and not the resource map. It therefore misses a newer package created by a prov only save, but only when the user is on a `/view/SpecificResourceMapPid` page. 

So to complete this issue, when navigating to the view page via a resource map PID, we need to check the resource map version chain so the viewer can show a link to the newer package.
